src/did.ts

```diff
@@ -304,9 +308,22 @@ export class DID {
       const issuerResolution = await this.resolve(issuerUrl)
       const controllerProperty = issuerResolution.didDocument?.controller
       const controllers = extractControllers(controllerProperty)
-      const signerIsController = signerDid ? controllers.includes(signerDid) : false
-      if (!signerIsController) {
-        throw new Error(`invalid_jws: not a valid verificationMethod for issuer: ${kid}`)
+
+      if (options.capability && options.capability.s) {
+        if (
+          (options.capability.p.iss === options.issuer ||
```

Comment:
Can we please check this condition first before we resolve the `issuerURL` DID resolution is expensive and with the current implementation we will always resolve the DID even if the cacao iss is equal to the options.issuer.